### PR TITLE
[Gang] The Hit Contract: The Non-Conversion Based Snowball Method.

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
:cl: Cobby
add: Influence can now be earned fairly rapidly by placing hits on people with the new hit contract. You get extra points if the target is a gang member, and even more if they are a gang head!
balance: Anyone who is a target of a hit cannot be converted by said gang.
balance: Contracts are tied to the gang that spawned them. If you are using another gang's contract, you can rack up points for them very easily. DON'T DO THIS! Just hide it from them, as they just lost 40 Influence for nothing.
/:cl:

**WARNING: This has NOT been tested to the fullest extent since it relies on multiple people being on at once. This was also "tested" in the morning hours so take that with a grain of salt.**

# Description
The Hit contract is a fairly high cost item/investment that allows for Rapid acquisition of points by killing particular people. It trades potential converts into blood-stained influence, with bonuses if they are in enemy gangs. Because of this, a new viable strategy is to rush tagging, save for the hit contract, then play ping pong and rack up points to give your entire [albeit small] gang tons of weaponry and ammo.

Targets are mind-based and won't appear again in the list of potential targets after a hit was placed on them by your gang, nor allow for conversion. This is to prevent exploits such as killing a non-gang target , cloning and repeat; killing a gang member, cloning, then penning them; or simply penning someone who you have a hit on. There are no tells for others if there's a hit on someone sans examining the document, so you as the gang leader have to coordinate with your team or you could get them outted as a leader.

The contract automatically checks to see if the target has died once a hit has been placed. It does not have to be at the gang's hands specifically, leaving room for beneficial accidents.

## Points Overview
Cost - 40
Target Is not a gang member or gets Deleted (bombs or badmins, or both!) - 50 [10 profit]
Target is a Gang Member, but not a boss - 65 [25 profit]
Target is a Gang Leader - 75 [35 profit]

As you can see, this always gives a full refund on a successful hit. This is not actual paper and will cost inventory slots, so you may have to hide the contracts later leaving room for a smart detective to stop you before you get too powerful.

# Reasoning
This was made as a way to deter conversion, which seemed to be a problem from the gang feedback thread. While you definitely want early conversion, mid-late game will now reward you for playing less conversion snowball [aka take 20 years for a gang round to end] and more aggressive PRECISE KILLING [if you murderbone you lose potential points] since conversion lowers your point potential while also increasing the point potential of others. Apart from that, you now have 2 general approaches you can take to gang: The Conversion ZergRush OR The High Octane Hit Squad, both being viable.

https://tgstation13.org/phpBB/viewtopic.php?f=10&t=9146#p264836

### CODE REVIEW

**Comment by Bawhoppen** [comment]
only skimming, I think the benefit of having more able hands will outweight the benefit of this

**Comment by XDTM** [comment]
I think gangs may need more buyable stuff if you want to encourage this, currently the expense is mostly pens and implant breakers, which are for conversion gameplay.

**Comment by Thunder12345** [comment]
Also, a small number of people with lots of guns and ammo are likely just going to get overrun by conversion gangs.

**Comment by KorPhaeron** [comment]
This doesnt seem at all viable. There isn't time to fuck around with this when you are dealing with conversion spam from two rival gangs + sec. Influence isnt actually a very important resource.

Only time I could see myself using this was if I had someone in crit or cuffs to cash in immediatley.

File: code/game/gamemodes/gang/gang_items.dm [hunk 1]
```diff
@@ -292,3 +293,88 @@
...
+		mind2bodylist += mind2body
+	for(var/datum/mind/PT in gang_ref.prev_targets) // CHEATER CHEATER PUMPKIN EATER
+		var/mob/prev_targ2body = PT.current
+		mind2bodylist += prev_targ2body
+
+	var/list/selection = GLOB.player_list - mind2bodylist
+
+	var/choice = input(user,"Who do you want dead?","Choose Your Victim") as null|anything in selection
+
+	if(!(isliving(choice)))
```
Comment (ChangelingRain) [inline code/game/gamemodes/gang/gang_items.dm]: you need to check if(target.stat == DEAD) as well.

File: code/game/gamemodes/gang/gang_items.dm [hunk 2]
```diff
@@ -292,3 +293,88 @@
...
+	for(var/datum/mind/M in gang_ref.gangsters) // You can only target people outside the gang that spawned the contract.
+		var/mob/mind2body = M.current
+		mind2bodylist += mind2body
+	for(var/datum/mind/PT in gang_ref.prev_targets) // CHEATER CHEATER PUMPKIN EATER
+		var/mob/prev_targ2body = PT.current
+		mind2bodylist += prev_targ2body
+
+	var/list/selection = GLOB.player_list - mind2bodylist
+
+	var/choice = input(user,"Who do you want dead?","Choose Your Victim") as null|anything in selection
```
Comment (ChangelingRain) [inline code/game/gamemodes/gang/gang_items.dm]: remember, this input is a sleep, so you need to make sure you can't spam-click the contract and that your selection exists afterwards.

**Comment by ChangelingRain** [comment]
Maybe if it was cheaper, yeah, or if they had more things to buy.

**Comment by coiax** [comment]
But influence isn't a particularly scarce resource? This seems a lot of bookkeeping for a very minor gain.

**Comment by Durkel** [comment]
People are only going to use this on people they've got cuffed or held down somewhere for free points.

**Comment by InsaneHyena** [comment]
You can't fix gang with little tweaks like that. The whole gamemode is bad and must be rebuilt from the ground up.

**Comment by Thunder12345** [comment]
It would work better if it was the ONLY way to play gang, instead of conversion it's preset teams, and the contract target is random, so you can't kidnap people THEN contract them.

**Comment by Iamgoofball** [comment]
# /advert HIT ACCEPTED

**Comment by ExcessiveUseOfCobblestone** [comment]
yeah you right it'll probably need a more thorough rework than simply an addition of an item.

**Pull request closed without merging**